libplatsupport/mach_include/spacemit/platsupport/mach/timer.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2025, 10xEngineers
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+#pragma once
+
+/* The K1 SoC has multiple timers, the 'Timer1' peripheral is used
+ * by this driver. Each timer peripheral consists of three individual
+ * counters/timers, confusingly named Timer 0, Timer 1 and Timer 2.
+ * Below Timer 0 is called TIMER0 and is used as a counter and Timer 1
+ * is called TIMER1 and used for timeouts.
+ */
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdbool.h>
+
+#define SPACEMIT_TIMER_BASE      0xD4014000
+#define APB_CLK_BASE             0xD4015000
+#define SPACEMIT_TIMER_SIZE      0x2000u
+#define SPACEMIT_TIMER0_IRQ      23u // timer0_1_irq: Timer 0 IRQ of Timer1
+#define SPACEMIT_TIMER1_IRQ      26u // timer1_1_irq: Timer 1 IRQ of Timer1
+#define SLOW_CLOCK false
+
+#if SLOW_CLOCK
+#define SPACEMIT_TIMER_TICKS_PER_SECOND  32768u
+#define K1_TCCR_CS0_VALUE                0x1u   /* slow clock */
+#else
+#define SPACEMIT_TIMER_TICKS_PER_SECOND  12800000u
+#define K1_TCCR_CS0_VALUE                0x0u   /* fast clock */
+#endif
+
+#define SPACEMIT_NUM_TIMERS 3
+#define SPACEMIT_TIMER_MAX_TICKS UINT32_MAX
+#define APBC_TIMERx_CLK_RST_OFFSET  0x34u
+
+typedef struct {
+    uint32_t tcer;         // Timer Count Enable Register
+    uint32_t tcmr;         // Timer Count Mode Register
+    uint32_t tcrr;         // Timer Count Restart Register
+    uint32_t tccr;         // Timer Clock Control Register
+
+    uint32_t tmr[3][4];    // Timer Match Register (TMRx)
+    uint32_t tplvr[4];     // Timer Preload Value Register (TPLVRx)
+    uint32_t tplcr[4];     // Timer Preload Control Register (TPLCRx)
+
+    uint32_t tier[4];       // Timer Interrupt Enable Register (TIERx)
+    uint32_t ticr[4];       // Timer Interrupt Clear Register (TICLRx)
+    uint32_t tsr[4];        // Timer Status Register (TSRx)
+
+    uint32_t tcnt[4];       // Timer Count Register (TCRx)
+} spacemit_timer_regs_t;
+
+typedef struct {
+    volatile spacemit_timer_regs_t *regs;
+    uint8_t  timer_n;
+    uint32_t value_h;
+    uint32_t *vaddr;
+} spacemit_timer_t;
```

Comment:
To make things nicer, I would get rid of this structure and move `value_h` into `spacemit_ltimer_t` instead, it's only used by the counter. And then pass timer_n as an argument to the common functions.